### DISCUSSION TITLE
[Selection for FlatTree] Is there a way to disable selection for some items?

### DISCUSSION THREAD

**Discussion by DanielMeng** [microsoft/fluentui#32834] [Q&A]
We would like to disable selection for some items in `FlatTree`. 
Wandering if this functionality is supported now or in plan.

**Comment by mainframev** [comment]
Hi @DanielMeng 👋🏻 

I'm curious about the use cases for having a disabled item in the tree. Would it perhaps make sense not to show it at all? 🤔 AFAIK, there are no plans to add that.

**Reply by DanielMeng** [reply]
couple of scenarios:
- multi selection mode, grey out certain items that are disabled. (controlled by something else outside of tree)
- single selection case, we are only interested in leaf nodes. And would like to hide radio button altogether from branches etc

**Reply by DanielMeng** [reply]
I was able to achieve above scenarios by manipulating the `selector` of `TreeItemLayout`.

**Reply by bitbanger** [reply]
> I was able to achieve above scenarios by manipulating the `selector` of `TreeItemLayout`.

Could you please provide an example or some detail on how you solved this? I'm very curious to do the same thing. I don't know why Fluent had such a big regression from v8 to v9 in terms of replicating the GroupedLists....

**Reply by RickSrick** [reply]
probably this way:
```jsx
<FlatTreeItem>
  <TreeItemLayout selector={{ disabled: true }}>{content}</TreeItemLayout>
</FlatTreeItem>
```
